src/v/kafka/server/handlers/list_partition_reassignments.cc

```diff
@@ -0,0 +1,205 @@
+            reassignments_by_topic[ntp.tp.topic].partitions.push_back(
+              std::move(partition_reassignment));
+        } else {
+            reassignments_by_topic[ntp.tp.topic] = ongoing_topic_reassignment{
+              .name = ntp.tp.topic, .partitions = {partition_reassignment}};
+        }
+    }
+
+    std::vector<ongoing_topic_reassignment> all_active_reassignments;
+    all_active_reassignments.reserve(reassignments_by_topic.size());
+    for (auto& [_, reassignments] : reassignments_by_topic) {
+        all_active_reassignments.push_back(std::move(reassignments));
+    }
+
+    return all_active_reassignments;
+}
+
+template<>
+ss::future<response_ptr> list_partition_reassignments_handler::handle(
+  request_context ctx, [[maybe_unused]] ss::smp_service_group ssg) {
+    list_partition_reassignments_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    list_partition_reassignments_response resp;
+
+    if (!is_controller(
+          ctx.metadata_cache().get_controller_leader_id(),
+          config::node().node_id())) {
+        vlog(klog.debug, "This node is not the controller.");
+        resp.data.error_code = error_code::not_controller;
+        resp.data.error_message = ss::sstring{
+          error_code_to_str(error_code::not_controller)};
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    if (!ctx.authorized(
+          security::acl_operation::describe, security::default_cluster_name)) {
+        vlog(
+          klog.debug,
+          "Failed cluster authorization. Requires DESCRIBE permissions on the "
+          "cluster.");
+        resp.data.error_code = error_code::cluster_authorization_failed;
+        resp.data.error_message = ss::sstring{
+          error_code_to_str(error_code::cluster_authorization_failed)};
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    // If topics is undefined, get all active reassignments
+    if (!request.data.topics.has_value()) {
+        auto all_active_reassignments = all_active_partition_reassignments(
+          ctx.metadata_cache().updates_in_progress());
+        resp.data.topics.reserve(all_active_reassignments.size());
+        std::for_each(
+          all_active_reassignments.begin(),
+          all_active_reassignments.end(),
+          [&resp](const ongoing_topic_reassignment& topic_reassignment) {
+              resp.data.topics.push_back(topic_reassignment);
+          });
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    resp.data.topics.reserve(request.data.topics->size());
+    for (const list_partition_reassignments_topics& tp : *request.data.topics) {
+        ongoing_topic_reassignment topic_reassignment{.name = tp.name};
+
+        for (const model::partition_id& pid : tp.partition_indexes) {
+            model::ntp ntp{model::kafka_namespace, tp.name, pid};
+
+            auto shard = ctx.shards().shard_for(ntp);
```

Comment:
we do not have to look for the shard, the reassignment information is managed centrally in topics_table, just accept it directly from this shard.

Reply:
This will not work as the partition may not be instantiated on the controller then it will not be in the shard table.